src/USER-HSMA/hsma2d.cpp

```diff
@@ -0,0 +1,2309 @@
+/* ----------------------------------------------------------------------
+ LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+ http://lammps.sandia.gov, Sandia National Laboratories
+ Steve Plimpton, sjplimp@sandia.gov
+
+ Copyright (2003) Sandia Corporation.  Under the terms of Contract
+ DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+ certain rights in this software.  This software is distributed under
+ the GNU General Public License.
+
+ See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Jiuyang Liang (liangjiuyang@sjtu.edu.cn)
+------------------------------------------------------------------------- */
+
+#include "hsma2d.h"
+
+#include "atom.h"
+#include "comm.h"
+#include "complex.h"
+#include "domain.h"
+#include "error.h"
+#include "force.h"
+#include "math.h"
+#include "math_special.h"
+#include "memory.h"
+#include "update.h"
+
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+#include<iomanip>
+#include <immintrin.h>
+#include<iostream>
+
+#if defined(_OPENMP)
+	#include<omp.h>
+#endif
+
+extern "C" {void lfmm3d_t_c_g_(double* eps, int* nsource, double* source, double* charge, int* nt, double* targ, double* pottarg, double* gradtarg, int* ier); }
+extern int fab(int n);
+extern int isfab(int m);
+
+using namespace LAMMPS_NS;
+using namespace std;
```

Comment:
We prefer not having a "using namespace std;" since there are occasional ambiguities between the LAMMPS_NS namespace and class internal types and variables (they may not be in your code directly but in the header files that you include). If all C++ library functions are prefixed with `std::` they can be easily identified (and it is not that much more to type).

Reply:
Deleted "using namespace std;"